### DISCUSSION TITLE
Issue with NumPy when running Detectron2: AttributeError: module 'numpy' has no attribute 'bool'

### DISCUSSION THREAD

**Discussion by cniq182** [facebookresearch/detectron2#5271] [Q&A]
Hello everyone,

I'm encountering an error while attempting to fine-tune a PointRend model in a Google Colab environment. I'm using Detectron2 for this task. This is the error:

_"AttributeError: module 'numpy' has no attribute 'bool'"_

 The specific error occurs during training and seems to be related to NumPy. Here's the relevant code snippet where the error occurs:

```
> from detectron2.engine import DefaultTrainer
> 
> cfg = get_cfg()
> # Merge pre-trained model configuration with PointRend
> cfg.merge_from_file("detectron2_repo/projects/PointRend/configs/InstanceSegmentation/pointrend_rcnn_R_50_FPN_3x_coco.yaml")
> 
> cfg.DATASETS.TRAIN = ("gambes",)
> cfg.DATASETS.TEST = ()
> cfg.DATALOADER.NUM_WORKERS = 2
> 
> cfg.MODEL.WEIGHTS = "detectron2://PointRend/InstanceSegmentation/pointrend_rcnn_R_50_FPN_3x_coco/164955410/model_final_edd263.pkl"
> cfg.SOLVER.IMS_PER_BATCH = 2  # This is the real "batch size" commonly known to deep learning people
> cfg.SOLVER.BASE_LR = 0.001 # pick a good LR
> cfg.SOLVER.MAX_ITER = 300    # 300 iterations seems good enough for this toy dataset; you will need to train longer for a practical dataset
> cfg.SOLVER.STEPS = []        # do not decay learning rate
> cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 4   # The "RoIHead batch size". 128 is faster, and good enough for this toy dataset (default: 512)
> cfg.MODEL.ROI_HEADS.NUM_CLASSES = 3  # only has one class (ballon). (see https://detectron2.readthedocs.io/tutorials/datasets.html#update-the-config-for-new-datasets)
> # NOTE: this config means the number of classes, but a few popular unofficial tutorials incorrect uses num_classes+1 here.
> 
> os.makedirs(cfg.OUTPUT_DIR, exist_ok=True)
> trainer = DefaultTrainer(cfg)
> trainer.resume_or_load(resume=False)
> trainer.train()
```

I've checked the NumPy and Detectron2 documentation, but I haven't been able to pinpoint the root cause of this problem. It seems to be related to a change in NumPy 1.20, but I'm unsure how to resolve it.

I've tried changing references from np.bool to bool in the code, as suggested in the NumPy documentation, but the error persists.

Has anyone encountered this issue before or have any suggestions on how to fix it? I would appreciate any guidance or advice you can offer.

Thanks in advance for your help!

**Comment by xk-maki** [comment]
Maybe you can use a  lower version of numpy，such as 1.23.1， or change np.bool to np.bool_ .
